src/object/obj_rpc.c

```diff
@@ -44,17 +44,17 @@ crt_proc_daos_key_desc_t(crt_proc_t proc, daos_key_desc_t *key)
 	int rc;
 
 	rc = crt_proc_uint64_t(proc, &key->kd_key_len);
-	if (rc != 0)
-		return -DER_HG;
+	if (unlikely(rc))
```

Comment:
I think this is something we want to consider and look at globally, I'm not aware that it's recommended to start adding it to code just yet.

Reply:
It's already used in many places. Especially it was used in PROC code optimization which already lands.

Reply:
Yes, it is recommended to use those in places that are frequently called (like the RPC serialization code in this case).